btcec/signature.go

```diff
@@ -396,3 +408,130 @@ func RecoverCompact(curve *KoblitzCurve, signature,
 
 	return key, ((signature[0] - 27) & 4) == 4, nil
 }
+
+// signRFC6979 generates a deterministic ECDSA signature according to RFC 6979 and BIP 62.
+func signRFC6979(privateKey *PrivateKey, hash []byte) (*Signature, error) {
+
+	privkey := privateKey.ToECDSA()
+	N := order
+	k := nonceRFC6979(privkey.D.Bytes(), hash)
+	inv := new(big.Int).ModInverse(k, N)
+	r, _ := privkey.Curve.ScalarBaseMult(k.Bytes())
+	r.Mod(r, N)
+
+	if r.Sign() == 0 {
+		return nil, errors.New("calculated R is zero")
+	}
+
+	e := hashToInt(hash, privkey.Curve)
+	s := new(big.Int).Mul(privkey.D, r)
+	s.Add(s, e)
+	s.Mul(s, inv)
+	s.Mod(s, N)
```

Comment:
Per signature spec, you should be checking if s is zero just as you do to see if r is zero:

http://en.wikipedia.org/wiki/Elliptic_Curve_Digital_Signature_Algorithm

Honestly, this is not likely, but if you're checking r, you should check s.